src/xrCore/Memory/xalloc.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include "xrCore/xrMemory.h"
+
+template <typename T>
+class xalloc
+{
+public:
+    using size_type = size_t;
+    using difference_type = std::ptrdiff_t;
+    using pointer = T*;
+    using const_pointer = const T*;
+    using reference = T&;
+    using const_reference = const T&;
+    using value_type = T;
+
+    template <class Other>
+    struct rebind
+    {
+        using other = xalloc<Other>;
+    };
+
+    pointer address(reference ref) const { return &ref; }
+    const_pointer address(const_reference ref) const { return &ref; }
+
+    xalloc() = default;
+    xalloc(const xalloc<T>&) = default;
+
+    template <class Other>
+    xalloc(const xalloc<Other>&)
+    {
+    }
+
+    template <class Other>
+    xalloc& operator=(const xalloc<Other>&)
+    {
+        return *this;
+    }
+
+    pointer allocate(const size_type n, const void* p = nullptr) const { return xr_alloc<T>(n); }
+    void deallocate(pointer p, const size_type /*n*/) const { xr_free(p); }
+    void deallocate(void* p, const size_type /*n*/) const { xr_free(p); }
```

Comment:
The prototypes of the 'deallocate' methods looks just the same. Do you really need such overloads? Won't it lead to ambiguity errors in compile time? Or is it a designed and expected behaviour?

Reply:
I really don't know why we need this. But this line was in the original allocator from GSC, I just did not remove it.

Reply:
Probably, this helps avoid some problems..